cloudferrylib/os/actions/cinder_database_manipulation.py

```diff
@@ -109,6 +111,75 @@ def _clean_data(data):
     return data
 
 
+class CinderStorageMigrationProgressView(object):
+
+    def __init__(self, src_volumes, dst_volumes, volumes_size_map):
+        self.src_volumes = copy.deepcopy(src_volumes)
+        self.dst_volumes = copy.deepcopy(dst_volumes)
+        self.failed_volumes = []
+        self.volumes_size_map = volumes_size_map
+        self.avg_speed_mb_s = 0
+        self.num_migrated = 0
+
+        self.total_volumes_size_b = 0
+        self.migrated_volumes_size_b = 0
+        self.failed_volumes_size_b = 0
+        for v_id, v_size in self.volumes_size_map.iteritems():
+            self.total_volumes_size_b += v_size
+            if v_id in self.dst_volumes:
+                self.migrated_volumes_size_b += v_size
+
+    def show_stats(self):
+        if len(self.dst_volumes) > 0:
+            LOG.info('Total number of volumes already '
+                     'migrated to destenation cloud: %d',
+                     len(self.dst_volumes))
+
+            LOG.info('Volumes already migrated:\n%s',
+                     '\n'.join([
+                         "%s(%s)" % (v['display_name'], v['id'])
+                         for v in self.dst_volumes])
+            )
+
+        LOG.info('Total number of volumes to be migrated: %d',
+                 len(self.src_volumes))
+        LOG.info('List of volumes to be migrated:\n%s',
+                 '\n'.join([
+                      "%s(%s)" % (v['display_name'], v['id'])
+                      for v in self.src_volumes])
+        )
+
+    def sync_migrated_volumes_info(self, volume, elapsed_time):
+        self.num_migrated += 1
+        volume_size_b = self.volumes_size_map[volume['id']]
+        self.migrated_volumes_size_b += volume_size_b
+
+        if elapsed_time == 0:
+            self.avg_speed_mb_s = 0
+        else:
+            self.avg_speed_mb_s = (volume_size_b/(1024.0**2))/elapsed_time
+
+    def sync_failed_volumes_info(self, volume):
+        self.failed_volumes_size_b += self.volumes_size_map[volume['id']]
+        self.failed_volumes.append(volume)
+
+    def show_progress(self):
+        total_volumes_size_gb = self.total_volumes_size_b/(1024.0**3)
+        migrated_volumes_size_gb = self.migrated_volumes_size_b/(1024.0**3)
+        failed_volumes_size_gb = self.failed_volumes_size_b/(1024.0**3)
+
+        LOG.info('Number of migrated images %d of %d. '
+                 'Volumes migrated %.1f%% and '
+                 'failed %.1f%% of %.1f GB total at %.1f MB/s.',
+                 self.num_migrated,
+                 len(self.src_volumes) + len(self.dst_volumes),
+                 migrated_volumes_size_gb/total_volumes_size_gb*100,
```

Comment:
Would be much better to calculate speeds dynamically as we did for image copy: `cloudferrylib/utils/sizeof_format.py`